packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -129,22 +137,31 @@ class TextMapPropagator {
   }
 
   _injectTraceparent (spanContext, carrier) {
-    if (!this._config.experimental.traceparent) return
+    if (!this._hasPropagationStyle('inject', 'tracecontext')) return
 
     const sampling = spanContext._sampling.priority >= AUTO_KEEP ? '01' : '00'
     const traceId = spanContext._traceId.toString(16).padStart(32, '0')
     const spanId = spanContext._spanId.toString(16).padStart(16, '0')
     carrier[traceparentKey] = `01-${traceId}-${spanId}-${sampling}`
   }
 
+  _hasPropagationStyle (mode, name) {
+    return this._config.tracePropagationStyle[mode].includes(name)
+  }
+
   _extractSpanContext (carrier) {
+    // TODO: Traceparent needs to be prioritized, but this is a breaking change
+    // until we support tracestate headers too, otherwise we would lose origin
+    // and tags header data.
     return this._extractDatadogContext(carrier) ||
       this._extractTraceparentContext(carrier) ||
       this._extractB3Context(carrier) ||
       this._extractSqsdContext(carrier)
   }
 
   _extractDatadogContext (carrier) {
+    if (!this._hasPropagationStyle('extract', 'datadog')) return null
```

Comment:
Instead of conditions in individual extractors, the functions should be called in the order that the config was set.

Reply:
Just saw the above comment, but let's still hardcode the order in `_extractSpanContext` instead so that we have to change the logic in a single place down the line.

Reply:
Yep, still working on restructuring for that.